vdom/tests/test_core.py

```diff
@@ -147,3 +117,8 @@ def test_component_disallows_children_kwargs():
     void = create_component('void', allow_children=False)
     with pytest.raises(ValueError, message='<void /> cannot have children'):
         void(children=div())
+
+def test_immutable():
+    comp = div("Hello")
+    with pytest.raises(AttributeError):
+        comp.children = "Nello"
```

Comment:
I think in some of the example notebooks I _might_ have done mutative things and we'll need to change those after this.

Reply:
You hadn't done anything mutating, but there were other incompatibilities I discovered. I'm gonna add tests for those I did discover.